src/main/java/com/epam/ta/reportportal/dao/LogRepositoryCustomImpl.java

```diff
@@ -130,13 +132,31 @@ public boolean hasLogs(Long itemId) {
 				.map(LOG_UNDER_RECORD_MAPPER);
 	}
 
+	@Override
+	public Map<Long, List<IndexLog>> findAllIndexUnderTestItemByLaunchIdAndTestItemIdsAndLogLevelGte(Long launchId, List<Long> itemIds, int logLevel) {
+		JTestItem parentItemTable = TEST_ITEM.as(PARENT_ITEM_TABLE);
+		JTestItem childItemTable = TEST_ITEM.as(CHILD_ITEM_TABLE);
+
+		return INDEX_LOG_FETCHER.apply(dsl.selectDistinct(LOG.ID, LOG.LOG_LEVEL, LOG.LOG_MESSAGE, parentItemTable.ITEM_ID.as(ROOT_ITEM_ID), CLUSTERS.INDEX_ID)
+				.on(LOG.ID)
+				.from(LOG)
+				.join(childItemTable)
+				.on(LOG.ITEM_ID.eq(childItemTable.ITEM_ID))
+				.join(parentItemTable)
+				.on(DSL.sql(childItemTable.PATH + " <@ " + parentItemTable.PATH))
```

Comment:
Please rewrite it to list of ids from PATH, instead of using path itself
PS. Also, if you want, you can refactoring rest places in this repository, which using this kind of filtering by path.